DIPs/DIP1000.md

```diff
@@ -87,29 +88,72 @@ the function.
 
 #### Reachability vs. lifetime
 
-For each value `v` within a program we define the notion of <i>reachability</i>
+##### Reachability
+
+For each value `v` within a program we define the notion of _reachability_
 denoted as `reachability(v)`, akin to the lexical extent through which the value
 can be accessed.
 
-- For an rvalue, the reachability is the expression within which it is used.
+- For an unnamed temporary (either lvalue such as `array[f()]` or rvalue such as
+`42 + f()`), the reachability is the largest grammatical expression within which
+it is used.
 - For a named variable in a scope, the reachability is the lexical scope of the variable per the language rules.
-- For a module-level variable, reachability is considered infinite. Notation: `reachability(v)` `=` `∞`.
+- For a module-level variable, reachability is considered infinite. Notation:
+`reachability(v)` `=` `∞`.
+
+For values with finite reachability, we denote their reachability as a tuple
+`M:B:E`, where:
+
+- `M` is a string that uniquely identifies the module within the program
+- `B` is an integral that is the index of the character at which reachability
+begins, as follows:
+  - For temporaries, `B` is the index of the first character of the largest
+expression in source code, and `E` is the index of the last character of the
+same largest expression
+  - For named values, `B` is the index of the `;` or `,` that ends the
```

Comment:
In the case of variables defined in `if` statements (e.g. `if (auto x = f())` ), the above is not technically correct. E.g. the `if` statement may not have curly braces at all and there's no `;`, nor `,` to end the `x` variable declaration in my example.